cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -209,12 +213,12 @@ private Scheduler getSchedulerForSchedule(Id.Program program, SchedulableProgram
                                                String scheduleName) throws NotFoundException {
     ApplicationSpecification appSpec = getStore().getApplication(program.getApplication());
     if (appSpec == null) {
-      throw new NotFoundException("application", program.getApplicationId());
+      throw new ApplicationNotFoundException(program.getApplicationId());
     }
 
     Map<String, ScheduleSpecification> schedules = appSpec.getSchedules();
     if (schedules == null || !schedules.containsKey(scheduleName)) {
-      throw new NotFoundException("schedule", scheduleName);
+      throw new ScheduleNotFoundException(scheduleName);
```

Comment:
Like I said above, the ``Scheduler`` interface doesn't throw any exception. Should that be changed?